DllImportGenerator/DllImportGenerator/TypeSymbolExtensions.cs

```diff
@@ -126,5 +129,10 @@ public static bool IsAutoLayout(this INamedTypeSymbol type, ITypeSymbol structLa
             }
             return type.IsReferenceType;
         }
+
+        public static TypeSyntax AsTypeSyntax(this ITypeSymbol type)
+        {
+            return SyntaxFactory.IdentifierName(type.ToDisplayString());
```

Comment:
The default `ToDisplayString()` method implementation returns the representation in the original source. The generated source might not have the same using directives, so we should use the fully-qualified symbol display format to get the full name so we don't need to copy the usings.

```suggestion
            return SyntaxFactory.ParseTypeName(type.ToDisplayString(SymbolDisplayFormat.FullyQualifiedFormat));
```

Reply:
I could have sworn I tried having usings / needing a full name.

Reply:
Ah, I see. The default format of `CSharpErrorMessageFormat` also has `SymbolDisplayTypeQualificationStyle.NameAndContainingTypesAndNamespaces`, so it was working.